scripts/getsamplesummary.py

```diff
@@ -46,35 +49,52 @@ def getsampleinfofromname(pars, sample):
       replies = dbc.generalquery( query )
   return replies
 
-def makelinks(samplename, lanedict):
+def makelinks(family_id, cust_name, sample_name, lanes):
     try:
-      os.makedirs(outputdir + samplename)
+      os.makedirs(os.path.join(outputdir, 'exomes', sample_name, 'fastq'))
     except OSError:
       pass
-    for entry in lanedict:
-      fclane = lanedict[entry].split("_")
-      print fclane
+    for entry in lanes:
+      fclane = lanes[entry].split("_")
+      print(fclane)
       fastqfiles = glob.glob(params['DEMUXDIR'] + "*" + fclane[0] + "*/Unalign*/Project_*/Sample_*" + 
-                            samplename + "_*/*L00" + fclane[2] + "*gz")
+                            sample_name + "[BF]_*/*L00" + fclane[2] + "*gz")
       for fastqfile in fastqfiles:
         nameparts = fastqfile.split("/")[len(fastqfile.split("/"))-1].split("_")
         date_fc = fastqfile.split("/")[6].split("_")[0] + "_" + fastqfile.split("/")[6][-9:]
-        newname = (nameparts[3][-1:] + "_" + date_fc + "_" + samplename + "_" + nameparts[2] +
+        newname = (nameparts[3][-1:] + "_" + date_fc + "_" + sample_name + "_" + nameparts[2] +
                    "_" + nameparts[4][-1:] + ".fastq.gz")
-        print fastqfile
-        print newname
+        print(fastqfile)
+        print(os.path.join(outputdir, 'exomes', sample_name, 'fastq', newname))
+        try:
+          os.symlink(fastqfile, os.path.join(outputdir, 'exomes', sample_name, 'fastq', newname))
+        except:
+          print("Can't create symlink for {}".format(sample_name))
         try:
-          os.symlink(fastqfile, outputdir + samplename + "/fastq/" + newname)
+          if cust_name != None and family_id != None:
+            mip_outdir = os.path.join(outputdir, cust_name, family_id, 'exomes')
+            link_source = os.path.join(outputdir, 'exomes', sample_name)
+            print('mkdir {}'.format(mip_outdir))
+
+            os.makedirs(mip_outdir)
+            os.symlink(link_source, os.path.join(mip_outdir, sample_name))
         except:
-          print "Can't create symlink for " + samplename
+          print("Can't create symlink for {} in MIP_ANALYSIS/cust".format(sample_name))
 
 smpls = getsamplesfromflowcell(params, fc)
 
 for sample in smpls.iterkeys():
+  family_id = None
+  cust_name = None
   with lims.limsconnect(params['apiuser'], params['apipass'], params['baseuri']) as lmc:
     pure_sample = sample.rstrip('BF') # remove the reprep (B) and reception control fail (F) letters from the samplename
     analysistype = lmc.getattribute('samples', pure_sample, "Sequencing Analysis")
     readcounts = .75 * float(analysistype[-3:])    # Accepted readcount is 75% of ordered million reads
+    family_id = lmc.getattribute('samples', pure_sample, 'familyID')
+    cust_name = lmc.getattribute('samples', pure_sample, 'customer')
+    if not re.match(r'cust\d{3}', cust_name):
```

Comment:
The linking will be skipped in make_link function and this should be put before make_link is even called. Now the dir structure will be created, which makes it a bit harder to catch this error just by glancing through the exomes dir. 
I'm actually all in to get emails when things like this happen, so we don't have to go through a pile of log files to see if something went wrong.

And for this particular error, we should also fix it in the LIMS as soon as it's caught :)